pgcli/packages/pgspecial.py

```diff
@@ -931,6 +932,26 @@ def show_help(cur, arg, verbose):  # All the parameters are ignored.
 def change_db(cur, arg, verbose):
     raise NotImplementedError
 
+def open_external_editor(cur, arg, verbose):
+    """
+    Open external editor, wait for the user to type in his query,
+    return the query.
+    :return: list with one tuple, query as first element.
+    """
+
+    filename = arg.strip().split(' ', 1)[0] if arg else None
+
+    MARKER = '# Type your query above this line.\n'
+    query = click.edit('\n\n' + MARKER, filename=filename)
+
+    if filename:
+        with open(filename, "r") as f:
```

Comment:
This will crash the app if the file cannot be opened due to permission issues. 

```
Traceback (most recent call last):
  File "/Users/amjith/.virtualenvs/pg-cli/bin/pgcli", line 9, in <module>
    load_entry_point('pgcli==0.16.3', 'console_scripts', 'pgcli')()
  File "/Users/amjith/.virtualenvs/pg-cli/lib/python2.7/site-packages/click/core.py", line 610, in __call__
    return self.main(*args, **kwargs)
  File "/Users/amjith/.virtualenvs/pg-cli/lib/python2.7/site-packages/click/core.py", line 590, in main
    rv = self.invoke(ctx)
  File "/Users/amjith/.virtualenvs/pg-cli/lib/python2.7/site-packages/click/core.py", line 782, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/amjith/.virtualenvs/pg-cli/lib/python2.7/site-packages/click/core.py", line 416, in invoke
    return callback(*args, **kwargs)
  File "/Users/amjith/Dropbox/code/python/pgcli/pgcli/main.py", line 383, in cli
    pgcli.run_cli()
  File "/Users/amjith/Dropbox/code/python/pgcli/pgcli/main.py", line 203, in run_cli
    sql = list(pgexecute.run(document.text))[0][0]
  File "/Users/amjith/Dropbox/code/python/pgcli/pgcli/pgexecute.py", line 193, in run
    for result in pgspecial.execute(cur, sql):
  File "/Users/amjith/Dropbox/code/python/pgcli/pgcli/packages/pgspecial.py", line 1019, in execute
    return command_executor(cur, arg, verbose)
  File "/Users/amjith/Dropbox/code/python/pgcli/pgcli/packages/pgspecial.py", line 948, in open_external_editor
    with open(filename, "r") as f:
IOError: [Errno 2] No such file or directory: u'/Users/abc.txt'
```